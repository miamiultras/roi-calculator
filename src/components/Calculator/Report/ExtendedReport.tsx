import React from 'react';

import Paragraph from '../../Paragraph/Paragraph';
import BackButton from '../../BackButton/BackButton';
import Table from '../Table/Table';
import { Service } from '../../../interfaces/service.interface';

const ExtendedReport = ({ extendedData }: { extendedData: Service[] }) => (
  <div className='container'>
    <h1>Live PPV Event ROI Calculator</h1>
    <Paragraph>
      Dear User, based on your entries we did the math for you:
   </Paragraph>
    <Table data={extendedData} />
    <BackButton name='start again' to='' />
  </div>
);

export default ExtendedReport;
