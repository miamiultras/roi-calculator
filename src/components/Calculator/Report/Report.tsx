import React from 'react';

import Paragraph from '../../Paragraph/Paragraph';
import NextButton from '../../NextButton/NextButton';
import BackButton from '../../BackButton/BackButton';
import Table from '../Table/Table';
import { data } from './data.mock';

const Report = () => (
  <div className='container'>
    <h1>Live PPV Event ROI Calculator</h1>
    <Table data={data} />

    <Paragraph>
      Do you want to understand the logic behind this calculation? Grab your results and
      we will explain you everything about conversion rates, customer care and operation fees.
   </Paragraph>

    <NextButton name='find out now' to='extended-report' />
    <BackButton name='start again' to='' />
    <BackButton name='back' to='step3' />
  </div>
);

export default Report;
