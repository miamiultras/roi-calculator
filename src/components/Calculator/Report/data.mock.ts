import { Service } from './service.interface';

export const data: Service[] = [
  {
    name: 'Revenues',
    homeGrown: '€12,000,000.00',
    cleeng: '€17,144,000.00',
  },
  {
    name: 'Costs',
    homeGrown: '€738,130.00',
    cleeng: '€1,080,395.08',
  },
  {
    name: 'Profits',
    homeGrown: '€11,261,870,00',
    cleeng: '€17,144,000,00',
  },
];

export const extendedData: Service[] = [
  {
    name: 'Site development',
    homeGrown: '€14,000.00',
    cleeng: '€0',
  },
  {
    name: 'Site maintance',
    homeGrown: '€2,800.0',
    cleeng: '€0',
  },
  {
    name: 'Recording & production costs',
    homeGrown: '€7,000.00',
    cleeng: '€7,000.00',
  },
  {
    name: 'Live streaming fees',
    homeGrown: '€3,000.00',
    cleeng: '€3,000.00',
  },
  {
    name: 'Expected quality',
    homeGrown: 'Mid',
    cleeng: 'Mid',
  },
  {
    name: 'Total fixed costs',
    homeGrown: '€26,800.00',
    cleeng: '€10,000.00',
  },
  {
    name: 'Traffic required',
    homeGrown: '42,857',
    cleeng: '42,857',
  },
  {
    name: 'Conversion rate',
    homeGrown: '7%',
    cleeng: '10%',
  },
];
