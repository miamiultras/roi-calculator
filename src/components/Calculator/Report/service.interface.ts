export interface Service {
  name: string;
  homeGrown: string;
  cleeng: string;
}
