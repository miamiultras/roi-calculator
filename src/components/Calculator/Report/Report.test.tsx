import * as React from 'react';
import { shallow } from 'enzyme';
import Report from './Report';

function setup() {
  const props = {};
  const wrapper = shallow(<Report data={[]} />);
  return {
    wrapper,
    props,
  };
}

describe('Report component', () => {
  it('should match snapshot if no change was intended', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
