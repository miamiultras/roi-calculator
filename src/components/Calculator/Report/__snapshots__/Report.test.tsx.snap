// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Report component should match snapshot if no change was intended 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Report
    data={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Live PPV Event ROI Calculator
        </h1>,
        <Table
          data={Array []}
        />,
        <Paragraph>
          Do you want to understand the logic behind this calculation? Grab your results and we will explain you everything about conversion rates, customer care and operation fees.
        </Paragraph>,
        <NextButton
          name="find out now"
          to="extended-report"
        />,
        <BackButton
          name="start again"
          to=""
        />,
        <BackButton
          name="back"
          to="step3"
        />,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Live PPV Event ROI Calculator",
        },
        "ref": null,
        "rendered": "Live PPV Event ROI Calculator",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Do you want to understand the logic behind this calculation? Grab your results and we will explain you everything about conversion rates, customer care and operation fees.",
        },
        "ref": null,
        "rendered": "Do you want to understand the logic behind this calculation? Grab your results and we will explain you everything about conversion rates, customer care and operation fees.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "find out now",
          "to": "extended-report",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "start again",
          "to": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "back",
          "to": "step3",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Live PPV Event ROI Calculator
          </h1>,
          <Table
            data={Array []}
          />,
          <Paragraph>
            Do you want to understand the logic behind this calculation? Grab your results and we will explain you everything about conversion rates, customer care and operation fees.
          </Paragraph>,
          <NextButton
            name="find out now"
            to="extended-report"
          />,
          <BackButton
            name="start again"
            to=""
          />,
          <BackButton
            name="back"
            to="step3"
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Live PPV Event ROI Calculator",
          },
          "ref": null,
          "rendered": "Live PPV Event ROI Calculator",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Do you want to understand the logic behind this calculation? Grab your results and we will explain you everything about conversion rates, customer care and operation fees.",
          },
          "ref": null,
          "rendered": "Do you want to understand the logic behind this calculation? Grab your results and we will explain you everything about conversion rates, customer care and operation fees.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "find out now",
            "to": "extended-report",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "start again",
            "to": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "back",
            "to": "step3",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
